### PR TITLE
Initial Jenkins S2I documentation

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@adellape @bparees PTAL

File: using_images/other_images/jenkins.adoc [hunk 1]
```diff
@@ -106,6 +106,65 @@ To use a persistent volume store, the cluster administrator must define a link:.
 
 Once selected, you must link:../../dev_guide/templates.html#creating-resources-from-a-template[instantiate] the template to be able to use Jenkins.
 
+== Using Jenkins as an Source-To-Image builder
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: s/an/a/

File: using_images/other_images/jenkins.adoc [hunk 2]
```diff
@@ -106,6 +106,65 @@ To use a persistent volume store, the cluster administrator must define a link:.
...
+
+If you want to customize the official OpenShift Jenkins image, you can use the
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: s/use the/use/

File: using_images/other_images/jenkins.adoc [hunk 3]
```diff
@@ -106,6 +106,65 @@ To use a persistent volume store, the cluster administrator must define a link:.
...
+Docker layering or you can use the image as a Source-To-Image builder.
+You can use Source-To-Image to copy your custom Jenkins Jobs definitions, additional
+plugins or replace the provided `config.xml` file with your own, custom, configuration.
+
+In order to include your modifications in Jenkins image, you need to have a Git
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: s/in Jenkins/in the Jenkins/

File: using_images/other_images/jenkins.adoc [hunk 4]
```diff
@@ -106,6 +106,65 @@ To use a persistent volume store, the cluster administrator must define a link:.
...
+repository, with the following directory structure:
+
+* `plugins` folder that contains binary Jenkins plugins you want to copy into Jenkins 
+* `plugins.txt` file that list the plugins you want to install (see the section above)
+* `configuration/jobs` folder that contains the Jenkins job definitions
+* `configuration/config.xml` file that contains your custom Jenkins configuration
+
+Note that the content of `configuration` folder will be copied into `/var/lib/jenkins`
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: s/into/into the/

File: using_images/other_images/jenkins.adoc [hunk 5]
```diff
@@ -106,6 +106,65 @@ To use a persistent volume store, the cluster administrator must define a link:.
 
 Once selected, you must link:../../dev_guide/templates.html#creating-resources-from-a-template[instantiate] the template to be able to use Jenkins.
 
+== Using Jenkins as an Source-To-Image builder
+
+If you want to customize the official OpenShift Jenkins image, you can use the
+Docker layering or you can use the image as a Source-To-Image builder.
+You can use Source-To-Image to copy your custom Jenkins Jobs definitions, additional
```
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: should link source-to-image here.  look at what we do in our other s2i image documentation for reference:
https://docs.openshift.org/latest/using_images/s2i_images/nodejs.html on how to talk about s2i and what to link to.

**Comment by mfojtik** [comment]
@bparees comments addressed, thanks!

**Comment by bparees** [comment]
looks good to me, ready for @openshift/team-documentation

**Comment by adellape** [comment]
@mfojtik @bparees Sorry for the delay, LGTM. Is this applicable for OSE starting in 3.1.0?

File: using_images/other_images/jenkins.adoc [hunk 6]
```diff
@@ -106,6 +106,66 @@ To use a persistent volume store, the cluster administrator must define a link:.
...
+      "git": {
+        "uri": "git://github.com/custom/repository.git"
+      }
+    },
+    "strategy": { <2>
+      "type": "Source",
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "jenkins-1-centos7:latest"
```
Comment (adellape) [inline using_images/other_images/jenkins.adoc]: I take it for OSE we should conditionalize this to instead just show `jenkins:latest`?

```
$ oc get istag -n openshift | grep jenkins

jenkins:1                                    registry.access.redhat.com/openshift3/jenkins-1-rhel7:latest                          2 weeks ago   3634ee8eec8ac268a86133e1ee560bdaa7453dd566e86c08e2402c82145d8d8f
jenkins:latest                               registry.access.redhat.com/openshift3/jenkins-1-rhel7:latest                          2 weeks ago   3634ee8eec8ac268a86133e1ee560bdaa7453dd566e86c08e2402c82145d8d8f
```

**Comment by bparees** [comment]
@adellape OSE starting in 3.2, I don't believe we've published a rhel image with this content yet.  @mfojtik can correct me if i'm wrong.

File: using_images/other_images/jenkins.adoc (hunk 6 above)
Comment (bparees) [inline using_images/other_images/jenkins.adoc]: actually it should just always say jenkins:latest, good catch.

it should also have "namespace: openshift"

Comment (adellape) [inline using_images/other_images/jenkins.adoc]: @mfojtik I'll merge after the above 2 items are addressed.

Comment (mfojtik) [inline using_images/other_images/jenkins.adoc]: @adellape fixed... sorry for this to take that long... I must overlook this.

**Comment by mfojtik** [comment]
@adellape @bparees I don't think we published rhel7 jenkins with s2i yet.

**Comment by bparees** [comment]
@mfojtik that's correct, it will go out w/ 3.2

**Comment by tnguyen-rh** [comment]
(fyi) tracking BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1303402

**Comment by adellape** [comment]
@mfojtik :+1: Thanks.

**Comment by tnguyen-rh** [comment]
Revhistory in https://github.com/openshift/openshift-docs/pull/1725

**Pull request merged**